plugin/src/main/kotlin/com/varabyte/kobweb/intellij/startup/KobwebPostStartupProjectActivity.kt

```diff
@@ -1,25 +1,106 @@
+    private class ImportListener(
+        private val project: Project,
+        private val syncRequestedNotification: KobwebNotificationHandle?
+    ) : ProjectDataImportListener {
+        override fun onImportStarted(projectPath: String?) {
+            // If an import is kicked off in an indirect way, we should still dismiss the sync popup.
+            syncRequestedNotification?.expire()
+        }
+
         override fun onImportFinished(projectPath: String?) {
+            project.kobwebPluginState = when (project.hasAnyKobwebDependency()) {
+                true -> KobwebPluginState.INITIALIZED
+                false -> KobwebPluginState.DISABLED
+            }
+
             // After an import / gradle sync, let's just clear the cache, which should get automatically rebuilt
             // as users interact with their code.
             project.service<KobwebProjectCacheService>().clear()
         }
     }
 
     override suspend fun execute(project: Project) {
+        if (project.hasAnyKobwebDependency() && project.kobwebPluginState == KobwebPluginState.DISABLED) {
+            project.kobwebPluginState = KobwebPluginState.UNINITIALIZED
+        }
+
+        val refreshProjectAction = ActionManager.getInstance().getAction("ExternalSystem.RefreshAllProjects") as? RefreshAllExternalProjectsAction
+        val syncRequestedNotification = if (refreshProjectAction != null && project.kobwebPluginState == KobwebPluginState.UNINITIALIZED) {
+            KobwebNotifier.notify(
+                project,
+                "The Kobweb plugin requires a one-time sync to enable functionality.",
+                "Sync Project",
+                NotificationType.WARNING
+            ) {
+                ActionUtil.invokeAction(refreshProjectAction, { dataId ->
+                    when {
+                        PlatformDataKeys.PROJECT.`is`(dataId) -> project
+                        else -> null
+                    }
+                }, ActionPlaces.NOTIFICATION, null, null)
+            }
+        } else null
+
         val messageBusConnection = project.messageBus.connect()
-        messageBusConnection.subscribe(ProjectDataImportListener.TOPIC, ImportListener(project))
+        messageBusConnection.subscribe(
+            ProjectDataImportListener.TOPIC,
+            ImportListener(project, syncRequestedNotification)
+        )
+
+        messageBusConnection.subscribe(AnActionListener.TOPIC, object : AnActionListener {
+            override fun beforeActionPerformed(action: AnAction, event: AnActionEvent) {
+                if (action === refreshProjectAction) {
+                    // Dismiss the sync popup no matter how the user executed the sync action, either directly through
+                    // the notification or by pressing the Gradle sync project button.
+                    syncRequestedNotification?.expire()
+                }
+            }
+        })
```

Comment:
Would the ImportListener not accomplish this already?

Reply:
I think you're right. This code was written before I had the import listener hide the popup. I'll remove it.